prisma/schema.sql

```diff
@@ -35,265 +48,371 @@ CREATE TYPE PhysicianStatus as ENUM('ACTIVE', 'INACTIVE');
   FOREIGN KEY(application_processing_id) REFERENCES application_processing(id)
 );
 
 -- Create permits table
 CREATE TABLE permits (
-  id                SERIAL PRIMARY KEY NOT NULL,
-  rcd_permit_id     INTEGER NOT NULL UNIQUE,
-  expiry_date       DATE NOT NULL,
-  receipt_id        INTEGER,
-  active            BOOLEAN NOT NULL DEFAULT true,
-  application_id    INTEGER NOT NULL UNIQUE,
-  applicant_id      INTEGER NOT NULL,
-  created_at        TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP,
-  updated_at        TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  rcd_permit_id INTEGER,
+  type PermitType NOT NULL,
+  expiry_date DATE NOT NULL,
+  active BOOLEAN NOT NULL DEFAULT true,
+  application_id INTEGER NOT NULL UNIQUE,
+  applicant_id INTEGER NOT NULL,
+  created_at TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  updated_at TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP,
 
+  PRIMARY KEY(rcd_permit_id),
   FOREIGN KEY(application_id) REFERENCES applications(id),
   FOREIGN KEY(applicant_id) REFERENCES applicants(id)
 );
 
 -- Create replacements table (subset of applications)
 CREATE TABLE replacements (
-  id                         SERIAL PRIMARY KEY NOT NULL,
-  reason                     ReasonForReplacement NOT NULL,
-  lost_timestamp             TIMESTAMPTZ,
-  lost_location              VARCHAR(255),
-  stolen_police_file_number  INTEGER,
-  stolen_jurisdiction        VARCHAR(255),
+  application_id INTEGER,
+
+  -- Reason for replacement
+  reason ReasonForReplacement NOT NULL,
+  lost_timestamp TIMESTAMPTZ,
+  lost_location VARCHAR(255),
+  stolen_police_file_number INTEGER,
+  stolen_jurisdiction VARCHAR(255),
   stolen_police_officer_name VARCHAR(255),
-  description                TEXT,
-  application_id             INTEGER UNIQUE NOT NULL,
-  created_at                 TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP,
-  updated_at                 TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  event_description TEXT,
+  created_at TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  updated_at TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  applicant_id INTEGER NOT NULL,
 
-  FOREIGN KEY(application_id) REFERENCES applications(id)
+  PRIMARY KEY(application_id),
+  FOREIGN KEY(application_id) REFERENCES applications(id),
+  FOREIGN KEY(applicant_id) REFERENCES applicants(id)
 );
 
 -- Create renewals table (subset of applications)
 CREATE TABLE renewals (
-  id                             SERIAL PRIMARY KEY NOT NULL,
-  uses_accessible_converted_van  BOOLEAN DEFAULT FALSE,
-  requires_wider_parking_space   BOOLEAN DEFAULT FALSE,
-  application_id                 INTEGER UNIQUE NOT NULL,
-  created_at                     TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP,
-  updated_at                     TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  application_id INTEGER,
+  receive_email_updates BOOLEAN NOT NULL,
+
+  -- Physician information
+  physician_first_name VARCHAR(255) NOT NULL,
```

Comment:
Nice separation of each of the application types! It looks good as is and will work. There's still a good amount of overlap between `renewals` and `new_applications`, so we could consider creating an additional table for these common fields (mostly just the physician info, receive email updates, and the van/parking space fields). I'm cool with it either way

Reply:
That's a good point. While that reduces repetition, I think it makes the code harder to write and maintain. For instance, when creating a renewal application, we would have to use Prisma nested create, and it would be harder to debug/access an application record as we would need to access 3+ tables. Let's keep it this way for now